ssh-server/main.go

```diff
@@ -113,38 +118,96 @@ func serviceChannel(newChannel ssh.NewChannel) error {
 	}
 	defer channel.Close()
 
+	allow := map[string]bool{"shell": true, "pty-req": true, "window-change": true}
+	resize := make(chan *pty.Winsize, 1)
+	defer close(resize)
+	done := make(chan error)
+	for {
+		select {
+		case err := <-done:
+			if err != nil {
+				return fmt.Errorf("running pty: %v", err)
+			}
+			return nil
+		case req, ok := <-requests:
+			if !ok {
+				return nil
+			}
+			if req.WantReply {
+				req.Reply(allow[req.Type], nil)
+			}
+			switch req.Type {
+			case "pty-req":
+				r := bytes.NewReader(req.Payload)
+				term, err := readString(r)
+				if err != nil {
+					return fmt.Errorf("parsing TERM environment variable value: %v", err)
+				}
+
+				if err := readWindowSize(r, resize); err != nil {
+					return fmt.Errorf("reading window size: %v", err)
+				}
+
+				go func() {
+					done <- runPTY(channel, term, resize)
+				}()
+			case "window-change":
+				if err := readWindowSize(bytes.NewReader(req.Payload), resize); err != nil {
+					return fmt.Errorf("reading window size: %v", err)
+					continue
+				}
+			}
+		}
+	}
+}
+
+func runPTY(channel ssh.Channel, term string, resize chan *pty.Winsize) error {
 	// Start the command with a pseudo-terminal.
-	shell, err := pty.Start(exec.Command("bash"))
+	cmd := exec.Command("bash")
```

Comment:
for another CL: but it would be good to allow the user to specify the command to run instead of just bash.  can you file an issue and link to it here?

Reply:
https://github.com/googlegenomics/pipelines-tools/issues/88